modules/java/native/java_machine.c

```diff
@@ -92,24 +92,92 @@ java_machine_unref(JavaVMSingleton *self)
     }
 }
 
+static GArray *
+_jvm_options_array_append(GArray *jvm_options_array, char *option_string)
+{
+  JavaVMOption opt;
+  opt.optionString = option_string;
+  return g_array_append_val(jvm_options_array, opt);
+}
+
+static gboolean
+_is_jvm_option_predefined(const gchar *option)
+{
+  static const gchar *predefined_options[] =
+  {
+    "Djava.class.path",
+    "Djava.library.path",
+    NULL
+  };
+
+  for (gint i = 0; predefined_options[i] != NULL; i++)
+    {
+      if (!strncmp(option, predefined_options[i], strlen(predefined_options[i])))
```

Comment:
I wouldn't use `strncmp()` when the length of the string can not be known explicitly (`strlen()` won't stop either).